python/cudf/cudf/core/column/interval.py

```diff
@@ -0,0 +1,91 @@
+# Copyright (c) 2018-2021, NVIDIA CORPORATION.
+import pyarrow as pa
+import cudf
+from cudf.core.column import StructColumn
+
+
+class IntervalColumn(StructColumn):
+    def __init__(
+        self,
+        dtype,
+        mask=None,
+        size=None,
+        offset=0,
+        null_count=None,
+        children=(),
+        closed="right",
+    ):
+
+        super().__init__(
+            data=None,
+            dtype=dtype,
+            mask=mask,
+            size=size,
+            offset=offset,
+            null_count=null_count,
+            children=children,
+        )
+        self._closed = closed
+
+    @property
+    def closed(self):
+        if self._closed in ["left", "right", "neither", "both"]:
```

Comment:
We should validate this on construction instead of when someone requests the property.

Reply:
Sounds good :D